### ISSUE TITLE
Metadata.User.id must be json encoded string

### ISSUE THREAD

**Issue by philkra** [elastic/apm-server#2127]
Howdy!

# Preamble
I ran into a either deserialisation or specification problem. The spec define the `id` field of the `Metadata.user` [schema](https://www.elastic.co/guide/en/apm/server/6.7/metadata-api.html#metadata-user-schema) that it can be of type `"type": ["string", "integer", "null"]`.

# Problem Description 
The `metadata.user.id` serialised as number represented json object (e.g. `"id": 123`) without the quotation marks as defined by the [JSON specification](https://tools.ietf.org/html/rfc7159#page-6) renders the apm-server to respond with  a `500: Internal Server Error`. But, when the `id` is serialized as a string (e.g. `"id": "123"`) the apm-server responds with `200`.

### Correct
```
"{"metadata": "service": {"name":"demo", "agent":{"name":"i-am-an-agent", "version":"6.7.0"}},"user":{"id":"12345","username":"foobar","email":"email@acme.com"}}}
```

### Failing
```
"{"metadata": "service": {"name":"demo", "agent":{"name":"i-am-an-agent", "version":"6.7.0"}},"user":{"id":12345,"username":"foobar","email":"email@acme.com"}}}
```

# APM Server Version
```
$ ./apm-server version
apm-server version 6.7.1 (amd64), libbeat 6.7.1 [5665711214eb3fcb956e7d1c61795e427bf97097 built 2019-04-02 14:55:06 +0000 UTC]
```

**Comment by graphaelli** [comment]
Hey @philkra!  Can you confirm these are the payloads being sent?  Both trigger a 400 Bad Request when trying to reproduce locally, as `service` should be an object nested under `metadata`.

Both of these are accepted in my testing:

numeric id:
```
{"metadata":"service":{"name":"demo","agent":{"name":"i-am-an-agent","version":"6.7.0"}},"user":{"id":12345,"username":"foobar","email":"email@acme.com"}}
```

string id:
```
{"metadata":"service":{"name":"demo","agent":{"name":"i-am-an-agent","version":"6.7.0"}},"user":{"id":"12345","username":"foobar","email":"email@acme.com"}}
```

**Comment by philkra** [comment]
@graphaelli , an error must have happened as I was trying to reduce the json to only the relevant data.

The following json results in `500 Internal Server Error`:

```
{"metadata":{"process":{"pid":8},"system":{"hostname":"99d905fd0a47","architecture":"x86_64","platform":"Linux"},"service":{"name":"demo","version":"2.6.23","agent":{"name":"apm-agent-php","version":"6.7.0"},"language":{"name":"php","version":"7.1.9"},"environment":"development","framework":{"name":"foo","version":"1.2.3"}},"user":{"id":12345,"username":"foobar","email":"email@acme.com"}}}
```

where as this json string with `user.id` quoted `"` is accepted:
```
{"metadata":{"process":{"pid":7},"system":{"hostname":"99d905fd0a47","architecture":"x86_64","platform":"Linux"},"service":{"name":"demo","version":"2.6.23","agent":{"name":"apm-agent-php","version":"6.7.0"},"language":{"name":"php","version":"7.1.9"},"environment":"development","framework":{"name":"foo","version":"1.2.3"}},"user":{"id":"12345","username":"foobar","email":"email@acme.com"}}}
```

**Comment by graphaelli** [comment]
Thanks for clarifying.  I have reproduced the behavior locally on 6.7.1, the issue is:

https://github.com/elastic/apm-server/blob/5665711214eb3fcb956e7d1c61795e427bf97097/model/metadata/user.go#L44-L45

This issue has been present since the introduction of the v2 intake in 6.5.  This issue is not present in 7.0 where it was fixed as part of the ECS migration in https://github.com/elastic/apm-server/pull/1783.  We should be able to backport that bit to fix this, thanks for sticking with the report.
